src/System.Collections/src/System/Collections/Generic/Queue.cs

```diff
@@ -392,20 +388,48 @@ public bool MoveNext()
             {
                 if (_version != _q._version) throw new InvalidOperationException(SR.InvalidOperation_EnumFailedVersion);
 
-                if (_index == -2)
-                    return false;
-
-                _index++;
+                // Instead of return true/false, we assign the result to a variable and
+                // return it at the end. This helps decrease code size, as currently the
+                // jit cannot do this for us and generates code for 3 returns.
+                bool result = false;
```

Comment:
@stephentoub OK, sure. I've done testing and omitting the `result` variable doesn't seem to change much, so the changes in this PR are now as minimal as possible.